bagel/utility.py

```diff
@@ -4,10 +4,38 @@
 import typer
 
 
+def file_encoding_error_message(input_p: Path) -> str:
+    """Return a message for when a file cannot be read due to encoding issues."""
+    return typer.style(
+        f"Failed to decode the input file {input_p}. "
+        "Please ensure that both your phenotypic .tsv file and .json data dictionary have UTF-8 encoding.\n"
+        "Tip: Need help converting your file? Try a tool like iconv (http://linux.die.net/man/1/iconv) or https://www.freeformatter.com/convert-file-encoding.html.",
+        fg=typer.colors.RED,
+    )
+
+
 def load_json(input_p: Path) -> dict:
     """Load a user-specified json type file."""
-    with open(input_p, "r") as f:
-        return json.load(f)
+    try:
+        with open(input_p, "r", encoding="utf-8") as f:
+            return json.load(f)
+    except UnicodeDecodeError as e:
+        # TODO: Refactor once https://github.com/neurobagel/bagel-cli/issues/218 is addressed
+        typer.echo(
+            file_encoding_error_message(input_p),
+            err=True,
+        )
+        raise typer.Exit(code=1) from e
+    except json.JSONDecodeError as e:
+        typer.echo(
+            typer.style(
+                f"The provided data dictionary {input_p} is not valid JSON. "
+                "Please provide a valid JSON file.",
+                fg=typer.colors.RED,
+            ),
```

Comment:
why can't we use  file_encoding_error_message here?

Reply:
`file_encoding_error_message` specifically provides a template for the error message that is returned when a non-UTF8 encoding is detected. Since the same error message format is used for both a non-UTF8 JSON and TSV (two separate validator functions), the convenience is used function to minimize code duplication.

The above error message is only emitted in one place in the code, which is when there is a JSONDecodeError for the input data dictionary. So it wouldn't make much sense to refactor that out into a separate function, since it would pretty much just be the same as `typer.style` itself.